compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -727,12 +716,7 @@ impl<'a, 'tcx> Visitor<'tcx> for BoundVarContext<'a, 'tcx> {
                         // Elided lifetimes are not allowed in non-return
                         // position impl Trait
                         let scope = Scope::TraitRefBoundary { s: self.scope };
-                        self.with(scope, |this| {
-                            let scope = Scope::Elision { s: this.scope };
-                            this.with(scope, |this| {
-                                intravisit::walk_item(this, opaque_ty);
-                            })
-                        });
+                        self.with(scope, |this| intravisit::walk_item(this, opaque_ty));
```

Comment:
For TAIT, a trait object necessarily inside generic arguments, that have their own `ObjectLifetimeDefault` rib.